internal/rundata/rundata.go

```diff
@@ -125,6 +138,21 @@ func Properties(ctx context.Context, resv *binding.Reservation) map[string]strin
 	return m
 }
 
+// readMetadataProto reads and unmarshals the metadata proto file.
+// If a metadata proto does not exist, returns a nil metadata and nil error.
+func readMetadataProto() (*mpb.Metadata, error) {
+	const metadataFilename = "metadata.textproto"
+	if _, err := os.Stat(metadataFilename); err != nil {
+		return nil, nil
```

Comment:
I worry that swallowing the error from os.Stat is a little too subtle and might end up swallowing unintended errors as well. Can we just use os.ReadFile and return any error to the caller? The caller would then use os.IsNotExist to explicitly ignore that particular error.

Reply:
Good idea, done.